src/pybind/mgr/dashboard/frontend/src/app/shared/forms/cd-validators.ts

```diff
@@ -103,6 +103,17 @@ export class CdValidators {
     );
   }
 
+  /**
+   * Validator that performs SSL certificate validation for rgw service.
+   * @returns {ValidatorFn} A validator function that returns an error map containing `pattern`
+   *   if the validation check fails, otherwise `null`.
+   */
+  static rgwCert(): ValidatorFn {
+    return Validators.pattern(
+      /^-----BEGIN RSA PRIVATE KEY-----(\n|\r|\f)((.+)?((\n|\r|\f).+)*)(\n|\r|\f)-----END RSA PRIVATE KEY-----[\n\r\f]-----BEGIN CERTIFICATE-----(\n|\r|\f)((.+)?((\n|\r|\f).+)*)(\n|\r|\f)-----END CERTIFICATE-----[\n\r\f]-----BEGIN CERTIFICATE-----(\n|\r|\f)((.+)?((\n|\r|\f).+)*)(\n|\r|\f)-----END CERTIFICATE-----[\n\r\f]*$/
```

Comment:
It seems here we're enforcing to put private key first and the the certificate (twice?), but provided there is not a standard defining the order, we should reach out consensus on what to allow i.e. here it is an example of a valid .pem file containing first certificate and then the private key:
https://github.com/rhcs-dashboard/ceph-dev/blob/master/docker/haproxy/dashboard.pem

Reply:
Unit tests to ensure it works as expected?